src/test/java/org/breedinginsight/services/impl/BreedingMethodServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+                "Method Names are Equal"
+        );
+        assertTrue(
+                breedingMethodService.isDuplicateName(method, testMethod),
+                "Method Names are Equal (switched order)"
+        );
+        assertTrue(
+                breedingMethodService.isDuplicateName(method, testMethodLowerCase),
+                "Method Names are Equal (one is LowerCase)"
+        );
+        testMethod.setId(method.getId());
+        assertFalse(
+                breedingMethodService.isDuplicateName(testMethod, method),
+                "The methods are the same method (not duplicate data)"
+        );
+
+
+        testMethod.setName("misatch Name");
+        testMethod.setId(UUID.randomUUID());
+        assertFalse(
+                breedingMethodService.isDuplicateName(testMethod, method),
+                "Method Names are not Equal"
+        );
+        testMethod.setName(null);
+        assertFalse(
+                breedingMethodService.isDuplicateName(testMethod, method),
+                "Method Names are not Equal (one is null)"
+        );
+
+        testMethod.setName("name");
+        method.setName(null);
+        assertFalse(
+                breedingMethodService.isDuplicateName(testMethod, method),
+                "Method Names are not Equal (the other is null)"
+        );
+    }
+
+    @Test
+    @SneakyThrows
+    public void isDuplicateAbbreviation() {
+        ProgramBreedingMethodEntity method = makeMethod("ABC", "ABC");
+        ProgramBreedingMethodEntity testMethod = makeMethod("mismatch_name", "ABC");
+        ProgramBreedingMethodEntity testMethodLowerCase = makeMethod("mismatch_name", "abc");
+        assertTrue(
+                breedingMethodService.isDuplicateAbbreviation(testMethod, method),
+                "Method Abbreviations are Equal"
+        );
+        assertTrue(
+                breedingMethodService.isDuplicateAbbreviation(method, testMethod),
+                "Method Abbreviations are Equal (switched order)"
+        );
+        assertTrue(
+                breedingMethodService.isDuplicateAbbreviation(testMethodLowerCase, method),
+                "Method Abbreviations are Equal (one is lower case)"
+        );
+        testMethod.setAbbreviation("misatch_abbreviation");
+        assertFalse(
+                breedingMethodService.isDuplicateAbbreviation(method, testMethod),
+                "Method Abbreviations are not equal."
+        );
+        testMethod.setAbbreviation(null);
+        assertFalse(
+                breedingMethodService.isDuplicateAbbreviation(method, testMethod),
+                "Method Abbreviations are not equal (one is null)."
+        );
+    }
+
+    // Helper Methods //
+    private ProgramBreedingMethodEntity makeMethod(String nameSuffix, String abbrevSuffix){
+        ProgramBreedingMethodEntity method = new ProgramBreedingMethodEntity();
+
+        method.setName("Name" + nameSuffix!=null? nameSuffix : "junk");
```

Comment:
I think the entirety of `"Name" + nameSuffix != null` is  being evaluated for the ternary operator, and will always be `true`. I would add parenthesis. I wish Java had a [null coalescing operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator)!
```suggestion
        method.setName("Name" + (nameSuffix != null ? nameSuffix : "junk"));
```

Reply:
Great Catch.  Thanks.
Done